### DISCUSSION TITLE
External source query parameter - bug or by design?

### DISCUSSION THREAD

**Discussion by KristofferBroberg** [Budibase/budibase#1583] [Help] [answered]
Hi all!

When adding a parameter to a query it is not possible to save the query without replacing default value. I.E you can't leave it blank and run a successful query = save query. My idea was to later assign a value to this parameter by binding an input field to it.

**Comment by KernOil** [comment]
I believe it is a bug I have been putting the default values as 0 or null to get it to work. the only problem with this is with the dates you have to put a date in as a default value. also you cant submit a blank date in a form and have the query run successfully that for sure is a bug.  I have asked them to fix a couple times still waiting on that fix. here is a screen shot of a couple of my values. 
![image](https://user-images.githubusercontent.com/80058238/120000485-b69a7c80-bf87-11eb-8ed9-6d03af7073dc.png)

**Comment by mjashanks** [comment, marked as answer]
Hey @KristofferBroberg and @AndrewJRamay . I have created an issue for this: https://github.com/Budibase/budibase/issues/1650.

Additionally, we are in the process of developing a much deeper SQL integration - see issues below. This would alleviate many of your issues, if using SQL. They are listed below, if you are interested. 

Basically, we are trying to automate most of the common `INSERT` `UPDATE` `SELECT` `DELETE` uses of SQL. You should be able to use SQL in the same way as you use Internal tables (aside from creating/editing of the tables themselves)

https://github.com/Budibase/budibase/issues/1642
https://github.com/Budibase/budibase/issues/1640

**Reply by KernOil** [reply]
@mjashanks in this new SQL update, will I be able to select an external table that the form will automatically write to just like I can select an internal table? so I will not be relying on writing a query, just selecting a DB table that could be internal or external. 

That would be an amazing upgrade!!

**Reply by shogunpurple** [reply]
@AndrewJRamay that's correct. We are building this functionality out right now, and it will be available in the coming weeks.

**Reply by KernOil** [reply]
HECK YES!! huge functionality / usability upgrade. took me (no formal education in tech)  a month to figure out the current way of writing to an external SQL DB. keep up the good work!!
